### ISSUE TITLE
A quicker way to clean app data than reinstalling

### ISSUE THREAD

**Issue by noomorph** [wix/Detox#3175]
**Is your feature request related to a problem? Please describe.**

Currently, it takes too long to clean the app data the way we usually do it via uninstalling and installing it again.

**Describe the solution you'd like**

Add some way to tell:

```js
await device.cleanAppData();
```

**Describe alternatives you’ve considered**

`device.installApp({ delete: true })`, but it's too slow.

**Additional context**

iOS: https://stackoverflow.com/questions/56228470/ios-simulator-how-to-clear-cache-of-an-app-instead-of-entire-phone
Android: ??

**Comment by d4vidi** [comment]
@noomorph please merge this with https://github.com/wix/Detox/issues/2537 🙏🏻

**Comment by d4vidi** [comment]
@jonathanmos does this in fact align with https://github.com/wix/Detox/issues/2534?

**Comment by jonathanmos** [comment]
@d4vidi Not exactly. We aren't providing an api to clear app data in #2534, rather we're clearing app data behind the scenes where possible to avoid reinstallation.

**Comment by d4vidi** [comment]
But technically `resetAppState()` is going to be a public API, wouldn't it be equivalent?
